util/env_posix.cc

```diff
@@ -427,12 +427,27 @@ class PosixEnv : public Env {
           return IOError(dir, errno);
       }
     }
+    auto name_max = pathconf(dir.c_str(), _PC_NAME_MAX);
+    if (name_max == -1) {
+      name_max = 255;
+    }
+    auto len = offsetof(dirent, d_name) + name_max + 1;
+    std::unique_ptr<char[]> entryp(new char[len]);
     struct dirent* entry;
-    while ((entry = readdir(d)) != nullptr) {
+    Status s;
+    for (;;) {
+      int rc = readdir_r(d, reinterpret_cast<dirent*>(entryp.get()), &entry);
```

Comment:
readdir_r() is deprecated since glibc 2.24 , see http://man7.org/linux/man-pages/man3/readdir_r.3.html and https://lwn.net/Articles/696469/